nilearn/maskers/surface_maps_masker.py

```diff
@@ -0,0 +1,415 @@
+"""Extract data from a SurfaceImage, using maps of potentially overlapping
+brain regions.
+"""
+
+import warnings
+
+import numpy as np
+from joblib import Memory
+from scipy import linalg
+from sklearn.base import BaseEstimator, TransformerMixin
+
+from nilearn import signal
+from nilearn._utils import fill_doc, logger
+from nilearn._utils.cache_mixin import CacheMixin, cache
+from nilearn._utils.class_inspect import get_params
+from nilearn.maskers._utils import (
+    check_same_n_vertices,
+    check_surface_data_ndims,
+    concatenate_surface_images,
+)
+from nilearn.surface import SurfaceImage
+
+
+@fill_doc
+class SurfaceMapsMasker(TransformerMixin, CacheMixin, BaseEstimator):
+    """Extract data from a SurfaceImage, using maps of potentially overlapping
+    brain regions.
+
+        .. versionadded:: 0.11.1
+
+    Parameters
+    ----------
+        maps_img : :obj:`~nilearn.surface.SurfaceImage`
+            Set of maps that define the regions. representative time course \
+            per map is extracted using least square regression. The data for \
+            each hemisphere is of shape (n_vertices/2, n_regions).
+
+        mask_img : :obj:`~nilearn.surface.SurfaceImage`, optional, default=None
+            Mask to apply to regions before extracting signals. Defines the \
+            overall area of the brain to consider. The data for each \
+            hemisphere is of shape (n_vertices/2, n_regions).
+
+        allow_overlap : :obj:`bool`, default=True
+            If False, an error is raised if the maps overlaps (ie at least two
+            maps have a non-zero value for the same voxel).
+
+        %(smoothing_fwhm)s
+            This parameter is not implemented yet.
+
+        %(standardize_maskers)s
+
+        %(standardize_confounds)s
+
+        %(detrend)s
+
+        high_variance_confounds : :obj:`bool`, default=False
+            If True, high variance confounds are computed on provided image \
+            with :func:`nilearn.image.high_variance_confounds` and default \
+            parameters and regressed out.
+
+        %(low_pass)s
+
+        %(high_pass)s
+
+        %(t_r)s
+
+        %(memory)s
+
+        %(memory_level1)s
+
+        %(verbose0)s
+
+        reports : :obj:`bool`, default=True
+            If set to True, data is saved in order to produce a report.
+
+        %(cmap)s
+            default="inferno"
+            Only relevant for the report figures.
+
+        clean_args : :obj:`dict` or None, default=None
+            Keyword arguments to be passed
+            to :func:`nilearn.signal.clean`
+            called within the masker.
+
+    Attributes
+    ----------
+        maps_img_ : :obj:`~numpy.ndarray`
```

Comment:
in nifti masker this is a nifti image, so I think it should be a surface image here:

https://nilearn.github.io/dev/modules/generated/nilearn.maskers.NiftiMapsMasker.html#nilearn-maskers-niftimapsmasker

![image](https://github.com/user-attachments/assets/9737d88f-93f9-4c7d-af76-4cc29d0933a2)

Reply:
But then there would be no difference between the fitted `maps_img_` and input `maps_img` and I would have to store the hemisphere-concatenated numpy array separately (leading to more memory usage).

Reply:
With surface maskers, I think the `.fit()` method would not be doing much except for input validation and SurfaceImage -> array conversions.

In nifti maskers, it is doing the resampling between mask and maps in addition.

Reply:
> But then there would be no difference between the fitted maps_img_ and input maps_img

That does necessarily shock me: I'd prefer to have some consistency across maskers in terms of the type of their attributes. Less cognitive load in general I think for devs and users.

I would do the concatenation across hemisphere at transform time.

Reply:
I see... So should we still return a fitted `self.maps_img_`?